src/main.rs

```diff
@@ -43,78 +41,72 @@ async fn event_handler(
     _: &Data,
 ) -> Result<(), Error> {
     match event {
-        serenity::FullEvent::Ready { data_about_bot, .. } => unsafe {
+        FullEvent::Ready { data_about_bot, .. } => {
             println!("Logged in as {}", data_about_bot.user.name);
             ctx.set_activity(Some(ActivityData::playing("Perfect Color!")));
 
-            if FIRST_TIME {
+            if unsafe { FIRST_TIME } {
```

Comment:
we could use a OnceLock (see https://doc.rust-lang.org/std/sync/struct.OnceLock.html#method.get_or_init) for the initialization instead to not need unsafe. That would also be thread-safe unlike the mutable static here.

Reply:
I like this idea, but I'd want to know what @Infiziert90 thinks as well. Not using unsafe should always be preferred, will fix tonight

Reply:
Is just a kind of flag to prevent from spamming the server with potential role creations, as OnReady can be triggered multiple times.
So what works best for it and is preferred

Reply:
~~Once~~Locked as resolved